### PR TITLE
Navigation

### LINKED ISSUE #2: Navigation
Build out the site nav + flyout containing the nav structure

### PR DESCRIPTION
**Pull request by brandonhundt**
closes #2 

- Adds the navbar
- creates button + menu item components
- Adds the navigation drawer
- initiates some other styles.

To Review:
- [ ] `$ bundle install`
- [ ] `$ bundle exec jekyll serve` (If you have problems, see the Readme)
- [ ] Go to http://127.0.0.1:4000/ in your browser
- [ ] Ensure the navbar is in place and is sticky to the top of the browser
- [ ] Open the menu drawer, ensure that works as expected, and closes when you click off.
- [ ] Note temporary nav items, once we have the sections in place, I'll create a loop to autogenerate the menu.
- [ ] Ignore the main content, will be addressed in a future PR.

### CODE REVIEW

File: assets/js/main.js
Comment (brandonhundt) [inline assets/js/main.js:1]: Note, I needed to remove last year's JS. Added Drawer init in this file

File: assets/js/carousel.js
Comment (brandonhundt) [inline assets/js/carousel.js:1]: Not needed.

File: _sass/_reset.scss
Comment (brandonhundt) [inline _sass/_reset.scss:1]: Since I opted out of a framework because this is pretty simple, I needed a reset. Found this.

**Comment by rpeterman-gp** [comment]
@brandonhundt We need to add a .tool-version file to this project.

File: .tool-versions
Comment (brandonhundt) [inline .tool-versions:1]: @rpeterman-gp added the tool-versions file.

**Comment by brandonhundt** [comment]
@rpeterman-gp I think this is more ready for you to look at. I've implemented the Popover API for the drawer and swapped out color SASS variables for CSS Variables. So there is no JS for this project as of yet.

Ready for more of your feedback and thoughts.

File: _sass/_menu-item.scss [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  &.yellow {
+    --_menu-link-background-color: #{$yellow};
+    --_menu-link-background-color--hover: #{mix(white, $yellow, 25%)};
+  }
+
+  &.light-blue {
+    --_menu-link-background-color: #{$cyan-a-10};
+    --_menu-link-background-color--hover: #{mix(white, $cyan, 25%)};
+  }
+}
```
Comment (brandonhundt) [inline _sass/_menu-item.scss:52]: @rpeterman-gp Got the variables set for this menu.

Comment (rpeterman-gp) [reply _sass/_menu-item.scss:52]: Oops. I goofed on which custom property should be change in the color variations. Should be setting the version w/o the underscore. I will fix the button styles to demonstrate.

File: _sass/_buttons.scss [hunk 2]
```diff
@@ -10,7 +10,7 @@
 
   background-color: var(--_btn-background-color);
 
-  color: $navy;
+  color: var(--navy);
```
Comment (brandonhundt) [inline _sass/_buttons.scss:13]: Should this be set to the `$navy`?

Comment (rpeterman-gp) [reply _sass/_buttons.scss]: Yep, but I am changing how that is wired up with the addition of the text color control properties.

**Pull request merged**